### PR TITLE
HW37 - NASA Mars Picture service

### PR DESCRIPTION
**Pull request by itsbohdan**

### CODE REVIEW

File: src/main/java/com/example/service/NasaPictureService.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        return photo;
+    }
+
+    private List<Photo> getCuriosityRoverPhotos(Integer sol, Optional<String> camera) {
+        try {
+            String nasaUrl = buildNasaUrl(sol, camera);
+            var response = restTemplate.getForEntity(nasaUrl, NasaResponse.class);
+
+            return response.getBody().photos();
```
Comment (StanislavZabramnyi) [inline src/main/java/com/example/service/NasaPictureService.java]: Possible NPE here, body can be null. I would do something like this 
`Optional.ofNullable(response.getBody())
                    .map(NasaResponse::photos).orElse(List.of());` . 
What do you think? 🤔

Comment (StanislavZabramnyi) [reply src/main/java/com/example/service/NasaPictureService.java]: In that case also would be good to add filter for not null values in the `findLargestPicture `

**Review by StanislavZabramnyi — APPROVED** [review]
Good job 👍  👏

File: src/main/java/com/example/controller/NasaPicturesController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+@RestController
+@RequestMapping("/mars/pictures")
+@RequiredArgsConstructor
+public class NasaPicturesController {
+
+    private final NasaPictureService nasaPictureService;
+
+    @GetMapping("/{sol}/largest")
+    public ResponseEntity<?> getLargestPicture(@PathVariable Integer sol, @RequestParam Optional<String> camera) {
```
Comment (szabramnyi) [inline src/main/java/com/example/controller/NasaPicturesController.java:23]: A wildcard in the return type is not a good practice. You can use ResponseEntity<Void> here, as the current response doesn't contain a body.

Comment (StanislavZabramnyi) [inline src/main/java/com/example/controller/NasaPicturesController.java]: A wildcard in the return type is not a good practice. You can use ResponseEntity here, as the current response doesn't contain a body.

Comment (StanislavZabramnyi) [reply src/main/java/com/example/controller/NasaPicturesController.java]: but, please pay attention to this item from the task description - "Configure the endpoint in a way that makes the browser to display the picture without changing the URL"

Comment (itsbohdan) [reply src/main/java/com/example/controller/NasaPicturesController.java]: updated to return content

**Pull request merged**